### PR TITLE
SALTO-6723 support elemID in fetch errors dummy adapter

### PR DESCRIPTION
**Pull request by omrifarkash**
---

---
_Release Notes_: 
_Dummy Adapter_:
- support elemID input for fetch errors

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/dummy-adapter/test/generator.test.ts [hunk 1]
```diff
@@ -16,293 +16,321 @@ import {
...
-    it('should create env variables if the env variable is set', async () => {
-      const envName = 'env'
-      process.env.SALTO_ENV = envName
-      const elements = await generateElements(testParams, mockProgressReporter)
-      expect(elements.find(e => e.elemID.getFullName() === 'dummy.envEnvObj')).toBeDefined()
-      expect(elements.find(e => e.elemID.getFullName() === 'dummy.envEnvObj.instance.envEnvInst')).toBeDefined()
-      expect(elements.find(e => e.elemID.getFullName() === 'dummy.EnvMeta')).toBeDefined()
-      expect(elements.find(e => e.elemID.getFullName() === 'dummy.EnvObj')).toBeDefined()
-      expect(elements.find(e => e.elemID.getFullName() === 'dummy.EnvObj.instance.EnvInst')).toBeDefined()
+  describe('generateFetchErrorsFromConfig', () => {
```
Comment (omrifarkash) [inline packages/dummy-adapter/test/generator.test.ts:312]: @IdoZakk This is the only test added

File: packages/dummy-adapter/src/generator.ts [hunk 2]
```diff
@@ -1024,3 +1033,15 @@ export const generateElements = async (
     ...envObjects,
   ].filter(e => !elementsToExclude.has(e.elemID.getFullName()))
 }
+
+export const generateFetchErrorsFromConfig = (
+  fetchErrorsFromConfig?: FetchErrorFromConfigFile[],
+): SaltoElementError[] | undefined => {
+  if (fetchErrorsFromConfig === undefined) {
+    return undefined
+  }
```
Comment (IdoZakk) [inline packages/dummy-adapter/src/generator.ts:1042]: You can replace this with a ? on the next line (after fetchErrorsFromConfig)

File: packages/dummy-adapter/src/generator.ts [hunk 3]
```diff
@@ -1024,3 +1033,15 @@ export const generateElements = async (
...
+  return fetchErrorsFromConfig.map(error => ({
+    elemID: ElemID.fromFullName(error.elemID),
+    ..._.omit(error, 'elemID'),
```
Comment (IdoZakk) [inline packages/dummy-adapter/src/generator.ts:1045]: You can do the other way around and you will not need to omit:
```typescript
return fetchErrorsFromConfig.map(error => ({
    ...error,
     elemID: ElemID.fromFullName(error.elemID)
```

**Review by IdoZakk — APPROVED** [review]
👍 
See my small comments

**Pull request merged**